frontend/js/src/home/Homepage.tsx

```diff
@@ -19,31 +19,42 @@
  */
 
 import * as React from "react";
-import { createRoot } from "react-dom/client";
-import * as Sentry from "@sentry/react";
-import { Integrations } from "@sentry/tracing";
-import { ToastContainer } from "react-toastify";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
 import { faSortDown, faSortUp } from "@fortawesome/free-solid-svg-icons";
 import { throttle } from "lodash";
-import { getPageProps } from "../utils/utils";
-import ErrorBoundary from "../utils/ErrorBoundary";
-import GlobalAppContext from "../utils/GlobalAppContext";
-import withAlertNotifications from "../notifications/AlertNotificationsHOC";
+import { useLoaderData, useNavigate } from "react-router-dom";
+import { Helmet } from "react-helmet";
 import NumberCounter from "./NumberCounter";
 import Blob from "./Blob";
+import GlobalAppContext from "../utils/GlobalAppContext";
 
 type HomePageProps = {
   listenCount: number;
   artistCount: number;
 };
 
-function HomePage({ listenCount, artistCount }: HomePageProps) {
+function HomePage() {
+  const { listenCount, artistCount } = useLoaderData() as HomePageProps;
   const homepageUpperRef = React.useRef<HTMLDivElement>(null);
   const homepageLowerRef = React.useRef<HTMLDivElement>(null);
+  const { currentUser } = React.useContext(GlobalAppContext);
+  const navigate = useNavigate();
 
   const [windowHeight, setWindowHeight] = React.useState(window.innerHeight);
 
+  React.useEffect(() => {
+    const redirectParam = new URLSearchParams(window.location.search).get(
+      "redirect"
+    );
+
+    if (
+      currentUser?.name &&
+      (redirectParam === "true" || redirectParam === null)
+    ) {
+      navigate(`/user/${currentUser.name}`);
+    }
+  }, [currentUser, navigate]);
```

Comment:
It doesn't quite look right to me that we would be handling the redirect here in this component.
Isn't there a way to do a conditional redirect in the route definition instead of here somehow?

Also the docs suggest using the `redirect` function exported from the react-router package instead of `navigate`

Reply:
I've now added a wrapper around the `Homepage` element. This element checks the search parameter and then decides if we want to load the element or redirect